Source/SIMPLib/CoreFilters/GenerateTiltSeries.cpp

```diff
@@ -0,0 +1,665 @@
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  // Generate Data Structure
+  std::pair<FloatArrayType::Pointer, ImageGeom::Pointer> gridPair = generateGrid();
+  if(nullptr == gridPair.first.get())
+  {
+    return;
+  }
+  FloatArrayType::Pointer gridCoords = gridPair.first;
+  ImageGeom::Pointer gridGeometry = gridPair.second;
+  SizeVec3Type gridDims = gridGeometry->getDimensions();
+
+  float increment = getIncrement();
+  size_t gridIndex = 0;
+
+  for(float currentDeg = 0.0f; currentDeg < 180.0f; currentDeg += increment)
+  {
+    QString gridDCName = QString(Detail::k_ImageGridNamePattern).arg(gridIndex);
+    DataContainer::Pointer gridDC = DataContainer::New(gridDCName);
+    ImageGeom::Pointer gridImageGeom = std::dynamic_pointer_cast<ImageGeom>(gridGeometry->deepCopy());
+    gridDC->setGeometry(gridImageGeom);
+
+    AttributeMatrix::Pointer cellAttr = AttributeMatrix::New({gridDims[0], gridDims[1], gridDims[2]}, Detail::k_AttributeMatrixName, AttributeMatrix::Type::Cell);
+    gridDC->insertOrAssign(cellAttr);
+
+    IDataArray::Pointer outputData = inputData->createNewArray(gridDims[0] * gridDims[1] * gridDims[2], {1}, getInputDataArrayPath().getDataArrayName());
+    cellAttr->insertOrAssign(outputData);
+    getDataContainerArray()->insertOrAssign(gridDC);
+
+    gridIndex++;
+  }
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateTiltSeries::preflight()
+{
+  // These are the REQUIRED lines of CODE to make sure the filter behaves correctly
+  setInPreflight(true); // Set the fact that we are preflighting.
+  emit preflightAboutToExecute(); // Emit this signal so that other widgets can do one file update
+  emit updateFilterParameters(this); // Emit this signal to have the widgets push their values down to the filter
+  dataCheck(); // Run our DataCheck to make sure everthing is setup correctly
+  emit preflightExecuted(); // We are done preflighting this filter
+  setInPreflight(false); // Inform the system this filter is NOT in preflight mode anymore.
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateTiltSeries::execute()
+{
+  initialize();
+  dataCheck();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  if(getCancel())
+  {
+    return;
+  }
+
+  // Compute a grid that forms a plane parallel to the rotation axis and centered halfway back in the volume
+  std::pair<FloatArrayType::Pointer, ImageGeom::Pointer> gridPair = generateGrid();
+  FloatArrayType::Pointer gridCoords = gridPair.first;
+  ImageGeom::Pointer gridGeometry = gridPair.second;
+  DataContainerArray::Pointer dca = getDataContainerArray();
+
+#ifdef SIMPL_USE_PARALLEL_ALGORITHMS
+  tbb::task_scheduler_init init;
```

Comment:
Can we use ParallelTaskAlgorithm here?  Branching code paths is a code smell, and thread count is already checked in its execute as long as the algorithm object is created before the for loop.    If SIMPL_USE_PARALLEL_ALGORITHMS is not supported, it is not performed in parallel.

Reply:
I don't think we can use ParallelTaskAlgorithm due to how the loops are developed with the rotations. I will continue to look at the code to see if we can shoe horn the algorithm into the ParallelTaskAlgorithm class.

Reply:
It should just be along the lines of 
```
ParallelTaskAlgorithm alg;
alg.execute(Detail::ResampleGrid(this, gridCoords, gridDC, rotationAxis, gridIndex);
```
due to the fact that both versions call Detail::ResampleGrid's operator() to execute. Then call alg.wait() at the end.  Just make sure that `alg` is defined before the loop so it can keep track of its own threads.

Reply:
If you can't use ParallelTaskAlgorithm, how can multi-threading be supported?  It's literally operating the same way as what's written in the loop?  Surely, they'd both have the same issue?